packages/test/test-version-utils/README.md

```diff
@@ -1,26 +1,137 @@
+
+- Use for tests that doesn't benefit or require any compat testing.
+- Test combination generated: [CompatKind.None]
+
+These compat `describe*` functions will also load the APIs with appropriate version and provide the test with a
+`TestObjectProvider` object, where the test can use to access Fluid functionality.  Even compat testing not
+necessary, `TestObjectProvider` provide convenience function and tracking capability, and it allows the test
+to enable compat testing easily in the future just by changing the `describe*`.
+
+### Legacy version defaults and installation
+
+By default, N-1, N-2, and LTS (hard coded) test variants are generated.  The versions can be specified using command
+line (see below) to run the test against any two versions. This package includes a `mocha` global hook that will
+install legacy packages at the beginning of the package based on the `compatVersion` settings.
+
+## Command line options
+
+Tests using the compat `describe*` will be controllable using the command line options when running mocha on
+driver selection, versions for compat testing, and compat kind combinations.
+
+```text
+--compatKind <CompatKind> - filter to the compat variant. See above table. Can specify multiple times.
+                                Default: undefined (no filter)
+--compatVersion <version> - specify the old version. Relative (to base) or specific version. Can specify multiple times.
+                                Default: -1, -2, LTS (hard coded in src/compatConfig.ts)
+--baseVersion <version>   - specify the base (new) version. Allow the test to run against any version combinations.
+                                Default: <current> (same version of the test package)
+--driver <driverType>     - <driverType> = "tinylicious" | "t9s" | "routerlicious" | "r11s" | "odsp" | "local"
+                                Default: "local"
+--r11sEndpointName <name> - Determine the environment variable name to look for r11s service information to run against
+                                Default: "r11s"
+--tenantIndex <number>    - The index into the tenant list to look for odsp server information to run against
+                                Default: 0
+--reinstall               - Force reinstall any required versions.  Default: reuse if a version is already installed.
+```
+
+`baseVersion` is a semver or a semver range.
+
+- If it is a semver range, the latest in that range will be picked.
+
+`compatVersion` can be a semver or semver range or an integer apply to the minor version relative to `baseVersion`
+
+- If it is a semver range, the latest in that range will be picked.
+- If it is an integer, the value will be apply to the minor version of the `baseVersion`, and create a `^` range
+including prerelease versions. The latest in that range will be picked.
+  - i.e. if `baseVersion` is `1.2.3`, and `compatVersion` is `-1`, the resulting range
```

Comment:
I thought ^1.1.3-0 would match anything up to 2.0, but that doesn't sound like what we want... Am I confused?

Reply:
I would expect it to be ~1.1.0 which I think will accept any 1.1.x.

Reply:
Yeah. I don't want to change the code for now, and it is not a problem right now with 0.x.  Probably need to fix this when we start using the major version.

Reply:
Then I would update the example here to be 0.x based, like the other example I saw elsewhere.